### PR TITLE
revamp v2

### PR DESCRIPTION
**Pull request by youliangtan**
## What's New in `v0.2.0`

- **Use the new `perform_action` for custom cleaning task**. During cleaning, the robot will act as a "readonly" robot, and update a short segment of the trajectory according to its current heading.
- **rmf map to robot map transformation:** Previously user is required to provide 2 sets of corresponding points. Now, with `RmfMapTransform` utils, users have the option to provide  `rmf_transform: [150.13, -24.846, 3.13, 0.05090] #[tx, ty, r, s]` in the yaml config file.
- **Location update and status update**: The updated fleet adapter provides more accurate location update to rmf, and more resilient towards locations that are not on the nav graph. Also, it will also constantly update the robot status to rmf-web, inform users about the current robot status (e.g. `uninitialized`, `offline`, `error`, `charging` ....) 
- **Mock testing of fleet adapter**: Use `--test_api_config_file` arg to test the ecobot fleet adapter in mock mode. This is helpful to test different scenarios on rviz before running on the actual robot.
- Major cleanups on fleet adapter implementation
- Create simple demo example to run `ecobot_fleet_adapter` in `office.world`

## Test fleet adapter with office world

- requires rmf_demos: https://github.com/open-rmf/rmf_demos/pull/140

Run this example in office world:
```
ros2 launch rmf_demos office.launch.xml run_fleet_adapters:=0
```
Then run the ecobot fleet adapter

```
ros2 run fleet_adapter_ecobot fleet_adapter_ecobot \
    -c src/fleet_adapter_ecobot/configs/robot_config.yaml \
    -n install/rmf_demos_maps/share/rmf_demos_maps/maps/office/nav_graphs/0.yaml \
    -s "ws://localhost:8000/_internal" \
    -tf src/fleet_adapter_ecobot/configs/test_api_config.yaml
```

For more info, please refer to the readme.

### CODE REVIEW

**Comment by Yadunund** [comment]
Is there a reason for removing sim_server.py?

File: fleet_adapter_ecobot/EcobotClientAPI.py [hunk 1]
```diff
@@ -55,7 +58,7 @@ def localize(self, init_point:str, map_name:str, rotate=False):
         if rotate: # The robot will rotate to improve localization
             url = self.prefix + f"/gs-robot/cmd/initialize?map_name={map_name}&init_point_name={init_point}"
         else: # The specified init point must be accurate
-            url = self.prefix + f"/gs-robot/cmd/initialize_directly?map_name={map_name}&init_point_name=?{init_point}"
```
Comment (Yadunund) [inline fleet_adapter_ecobot/EcobotClientAPI.py:58]: Did the endpoint in the API change? Or what the original incorrect?

**Review by Yadunund — COMMENTED** [review]
@youliangtan The changes here are okay and I trust they have been working well on hardware?

Wondering why `ecobot_sim_server.py` was removed? How will users control robots in simulation without this?

File: fleet_adapter_ecobot/EcobotClientAPI.py (hunk 1 above)
Comment (youliangtan) [reply fleet_adapter_ecobot/EcobotClientAPI.py:58]: The original was incorrect, tested with this and it works.

**Comment by youliangtan** [comment]
The `ecobot_sim_server.py` was removed since the current way I am testing the `ecobot fleet adapter` is through solely rviz (without simulation and with wall realtime). The `TestAPIClient.py` is activated when user add the `-tf` arg when running the script. This will mock the ecobot moving around the environment. You can try out by follow the instructionss above. 

However Iam still pondering whether to re-add and update the `ecobot_sim_server.py`, since there will be added benefits when testing the fleetadapter in full simulation.

**Comment by Yadunund** [comment]
My recommendation is to definitely keep and it in fact update `rmf_demos/office` to add a cleaning robot that can be commanded by this adpater. 

So along with the change in this PR https://github.com/open-rmf/rmf_demos/pull/140, update the building.yaml to add a cleaning robot which can either be controlled by `rmf_demos_fleet_adapter` or with  a combination of `fleet_adapter_ecobot` + `ecobot_sim_server.py`

**Comment by Yadunund** [comment]
Or maybe add a new package to this repo that will contain a copy of office map, launch files etc to control a robot in simulation using this adapter

**Review by Yadunund — APPROVED** [review]
Long overdue merge.

Many thanks @youliangtan for this huge enhancement!

**Comment by youliangtan** [comment]
Thanks for merging in, and also the fixes from @JohnTGZ :pray:

**Pull request merged**